### PR TITLE
[Merged by Bors] - Fix attestation withdrawals root mismatch

### PR DESCRIPTION
**Pull request by jimmygchen**
## Issue Addressed

Addresses #4234 

## Proposed Changes

- Skip withdrawals processing in an inconsistent state replay. 
- Repurpose `StateRootStrategy`: rename to `StateProcessingStrategy` and always skip withdrawals if using `StateProcessingStrategy::Inconsistent`
- Add a test to reproduce the scenario

### CODE REVIEW

File: beacon_node/beacon_chain/tests/attestation_verification.rs [hunk 1]
```diff
@@ -998,6 +1042,100 @@ async fn attestation_that_skips_epochs() {
...
+        .extend_chain(
+            // To trigger the bug we need the proposer attestation reward to be signed at a block
+            // that isn't the first in the epoch.
+            MainnetEthSpec::slots_per_epoch() as usize + 1,
+            BlockStrategy::OnCanonicalHead,
+            AttestationStrategy::SomeValidators(attesters),
+        )
+        .await;
+
+    // Add BLS change for the block proposer at slot 33. This sets up a withdrawal for the block proposer.
```
Comment (michaelsproul) [inline beacon_node/beacon_chain/tests/attestation_verification.rs:1065]: Clever approach!

File: consensus/state_processing/src/block_replayer.rs [hunk 2]
```diff
@@ -60,13 +60,13 @@ impl From<BlockProcessingError> for BlockReplayError {
     }
 }
 
-/// Defines how state roots should be computed during block replay.
-#[derive(PartialEq)]
-pub enum StateRootStrategy {
+/// Defines how state roots should be computed and whether to perform all state transitions during block replay.
+#[derive(PartialEq, Clone)]
```
Comment (michaelsproul) [inline consensus/state_processing/src/block_replayer.rs]: May as well make it `Copy`

```suggestion
#[derive(PartialEq, Clone, Copy)]
```

File: consensus/state_processing/src/block_replayer.rs [hunk 3]
```diff
@@ -263,6 +266,7 @@ where
                 &mut self.state,
                 block,
                 self.block_sig_strategy,
+                self.state_processing_strategy.clone(),
```
Comment (michaelsproul) [inline consensus/state_processing/src/block_replayer.rs]:
```suggestion
                self.state_processing_strategy,
```

**Review by michaelsproul — APPROVED** [review]
This is great! I can't fault it

Just one tiny suggestion and then I'm happy to merge

File: consensus/state_processing/src/block_replayer.rs (hunk 2 above)
Comment (jimmygchen) [reply consensus/state_processing/src/block_replayer.rs]: Nice tip!! 🙏

**Pull request closed without merging**